### PR TITLE
[vlc] Initial integration

### PR DESCRIPTION
**Pull request by AdamKorcz**
VLC is a free and open source cross-platform multimedia player and framework that plays most multimedia files as well as DVDs, Audio CDs, VCDs, and various streaming protocols.

It is used by many people across the globe - most likely including you - to watch and listen to great content.

-------------------------------------------------------------------

@tguillem @robUx4 @fcartegnie 

Are you interested in integrating VLC with OSS-fuzz?

This PR adds the build files and a working fuzzer.

To complete the integration, a maintainers email address is needed for bug reports.

### CODE REVIEW

**Comment by tguillem** [comment]
Hello @AdamKorcz 

This is really interesting, but you should use the fuzzers target we are already using:

```
test/vlc-demux-libfuzzer.c
test/src/input/demux-run.c
```

This target test our demuxers, packetizers, and decoders: the critical code that do input parsing.

I'm not very keen on pushing the OSS-fuzz integration with only the xml test (it should use also vlc-demux-libfuzzer.c).

Furthermore, could you propose the xml fuzzer test in the VLC codebase ? I think it's better to keep them all in one place. In the future, these tests could also be built from a distcheck to make sure we don't break the API.

**Comment by DavidKorczynski** [comment]
Friendly ping @tguillem - if you are happy to get VLC in I think we should be good to go when we have emails for the bug reports. @AdamKorcz and I are happy to push VLC forward with more fuzzers.

**Comment by tguillem** [comment]
Hello, sorry I missed your comment one week ago.

I have some minor comments (nitpicking):
 - Could you squash/fixup all your commits ? You are adding commits that are modifying/removing commits inside your PR
 - Is it possible to propose your configure fixups into vlc (https://code.videolan.org/videolan/vlc) ?

I can't decide alone the inclusion of VLC into OSS-fuzz, I'm pinging my team.

**Comment by DavidKorczynski** [comment]
> * Could you squash/fixup all your commits ? You are adding commits that are modifying/removing commits inside your PR

Yep, we usually squash when merging the PR. 

> * Is it possible to propose your configure fixups into vlc (https://code.videolan.org/videolan/vlc) ?

The current changes are oss-fuzz specific and not general issues, i.e. they are needed because of the OSS-Fuzz environment. I would be happy to suggest changes upstream but it really should only be done if VLC is interested in OSS-Fuzz as otherwise there is no need to do OSS-Fuzz-specific changes. 

> I can't decide alone the inclusion of VLC into OSS-fuzz, I'm pinging my team.

Sounds good, please let us know when you have something conclusive.

**Comment by jbkempf** [comment]
> I can't decide alone the inclusion of VLC into OSS-fuzz, I'm pinging my team.

Please merge and activate this inclusion.
LGTM.

**Comment by tguillem** [comment]
> Friendly ping @tguillem - if you are happy to get VLC in I think we should be good to go when we have emails for the bug reports. @AdamKorcz and I are happy to push VLC forward with more fuzzers.

Please use the following mail: ossfuzz@videolan.org

**Comment by DavidKorczynski** [comment]
@inferno-chromium @jonathanmetzman @asraa @oliverchang Calling on the current sheriff, this one is good to go!

File: projects/vlc/build.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Use OSS-Fuzz environment rather than hardcoded setup.
```
Comment (inferno-chromium) [inline projects/vlc/build.sh:18]: Can we file a bug to ask them to create a special makefile target or configure flag for all of these.

**Review by inferno-chromium — APPROVED** [review]
Amazing!

**Comment by DavidKorczynski** [comment]
@jbkempf @tguillem Thank you for the feedback here. Adam and I will continue to work on this one to improve so you should see upstream pull requests from us soon!

File: projects/vlc/build.sh (hunk 1 above)
Comment (DavidKorczynski) [reply projects/vlc/build.sh:18]: I will take care of this myself - please see comment https://github.com/google/oss-fuzz/pull/5598#issuecomment-843449990

**Pull request merged**